webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TraitTestApiFromYamlFiles.scala

```diff
@@ -0,0 +1,341 @@
+import com.normation.rudder.rest.lift.LiftApiProcessingLogger
+import com.normation.rudder.rest.lift.LiftHandler
+
+import com.normation.errors.IOResult
+import com.normation.errors.RudderError
+import com.normation.errors.effectUioUnit
+
+import scala.jdk.CollectionConverters._
+import org.specs2.mutable._
+import org.specs2.specification.core.Fragment
+import org.specs2.specification.core.Fragments
+import org.yaml.snakeyaml.Yaml
+import net.liftweb.common.Box
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Failure
+import net.liftweb.common.Full
+import net.liftweb.http.InMemoryResponse
+import net.liftweb.http.LiftResponse
+import net.liftweb.http.LiftRules
+import net.liftweb.mocks.MockHttpServletRequest
+import net.liftweb.util.Helpers.tryo
+
+import java.nio.charset.StandardCharsets
+import scala.util.control.NonFatal
+
+import zio._
+import zio.syntax._
+import com.normation.zio._
+import java.io.InputStream
+import java.nio.file.Path
+import java.nio.file.Paths
+
+import com.normation.errors._
+
+/*
+ * Utily data structures
+ */
+sealed trait ResponseType
+final object ResponseType {
+  final object Text extends ResponseType
+  final object Json extends ResponseType
+}
+
+// Our object representing a test specification
+final case class TestRequest(
+    description    : String
+  , method         : String
+  , url            : String
+  , queryBody      : String
+  , headers        : List[String]
+  , params         : List[(String, String)]
+  , responseType   : ResponseType
+  , responseCode   : Int
+  , responseContent: String
+)
+
+
+object TraitTestApiFromYamlFiles {
+
+  def buildLiftRules[A <: LiftApiModuleProvider[_ <: EndpointSchema]](modules: List[A], versions: List[ApiVersion], userService: Option[UserService]) = {
+    implicit val userServiceImp = userService match {
+      case None => new UserService {
+                      val user = new User {
+                        val account = RudderAccount.User("test-user", "pass")
+                        def checkRights(auth : AuthorizationType) = true
+                        def getApiAuthz = ApiAuthorization.allAuthz
+                      }
+                      val getCurrentUser = user
+                    }
+      case Some(u) => u
+    }
+
+    val apiDispatcher = new RudderEndpointDispatcher(LiftApiProcessingLogger)
+    val apiAuthorizationLevelService = new DefaultApiAuthorizationLevel(LiftApiProcessingLogger)
+
+    //append to list all new format api to test it
+    val rudderApi = new LiftHandler(apiDispatcher, versions, new AclApiAuthorization(LiftApiProcessingLogger, userServiceImp, () => apiAuthorizationLevelService.aclEnabled), None)
+    modules.foreach { module =>
+      rudderApi.addModules(module.getLiftEndpoints())
+    }
+
+    val liftRules = new LiftRules()
+    liftRules.statelessDispatch.append(rudderApi.getLiftRestApi())
+    (rudderApi, liftRules)
+  }
+
+}
+
+trait TraitTestApiFromYamlFiles extends Specification {
```

Comment:
the new trait to extends to automatically be able to exec yaml tests in a given directory in class path.